server/datastore/mysql/software.go

```diff
@@ -1721,3 +1721,307 @@ func (ds *Datastore) ListCVEs(ctx context.Context, maxAge time.Duration) ([]flee
 
 	return result, nil
 }
+
+func (ds *Datastore) ListHostSoftware(ctx context.Context, hostID uint, includeAvailableForInstall bool, opts fleet.ListOptions) ([]*fleet.HostSoftwareWithInstaller, *fleet.PaginationMetadata, error) {
+	// `status` computed column assumes that all results (pre, install and post)
+	// are stored at once, so that if there is an exit code for the install
+	// script and none for the post-install, it is because there is no
+	// post-install.
+	const stmtInstalled = `
+		SELECT
+			st.id,
+			st.name,
+			st.source,
+			si.filename as package_available_for_install,
+			hsi.created_at as last_install_installed_at,
+			hsi.execution_id as last_install_install_uuid,
+			CASE
+				WHEN hsi.post_install_script_exit_code IS NOT NULL AND
+					hsi.post_install_script_exit_code = 0 THEN ? -- installed
+
+				WHEN hsi.post_install_script_exit_code IS NOT NULL AND
+					hsi.post_install_script_exit_code != 0 THEN ? -- failed
+
+				WHEN hsi.install_script_exit_code IS NOT NULL AND
+					hsi.install_script_exit_code = 0 THEN ? -- installed
+
+				WHEN hsi.install_script_exit_code IS NOT NULL AND
+					hsi.install_script_exit_code != 0 THEN ? -- failed
+
+				WHEN hsi.pre_install_query_output IS NOT NULL AND
+					hsi.pre_install_query_output = '' THEN ? -- failed
+
+				WHEN hsi.host_id IS NOT NULL THEN ? -- pending
+
+				ELSE NULL -- not installed from Fleet installer
+			END AS status,
```

Comment:
Yeah , another option that I thought about doing but decided to wait to see how/where we need it would be to extract that sql part in a function, like we do for other bits of complex+reusable select.